tests/checks/integrity/outlier_detection_test.py

```diff
@@ -0,0 +1,99 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Contains unit tests for the outlier_detection check."""
+import numpy as np
+import pandas as pd
+from hamcrest import assert_that, calling, raises
+from sklearn.datasets import load_iris
+
+from deepchecks.tabular.checks import OutlierDetection
+from deepchecks.tabular.dataset import Dataset
+
+
+def test_condition_input_validation():
+    # Assert
+    assert_that(calling(OutlierDetection().add_condition_not_more_outliers_than).with_args(max_outliers_ratio=-1),
+                raises(ValueError,
+                       'max_outliers_ratio must be between 0 and 1'))
+
+
+def test_integer_single_column_no_nulls():
+    # Arrange
+    data = {'col1': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1000]}
+    dataframe = pd.DataFrame(data=data)
+    # Act
+    result = OutlierDetection(n_to_show=1, num_nearest_neighbors=5).run(dataframe)
+    # Assert
+    assert_that(max(result.value[0]) > 0.7)
```

Comment:
the "hamcrest way" to do that would be
`assert_that(result.value[0], has_item(greater_than(0.7))`